x/rollapp/keeper/msg_server_create_rollapp.go

```diff
@@ -49,3 +38,32 @@ func (k msgServer) CreateRollapp(goCtx context.Context, msg *types.MsgCreateRoll
 
 	return &types.MsgCreateRollappResponse{}, nil
 }
+
+func (k msgServer) checkIfRollappExists(ctx sdk.Context, msg *types.MsgCreateRollapp) error {
+	rollappId, err := types.NewChainID(msg.RollappId)
+	if err != nil {
+		return err
+	}
+	// check to see if the RollappId has been registered before
+	if _, isFound := k.GetRollapp(ctx, rollappId.GetChainID()); isFound {
+		return types.ErrRollappExists
+	}
+	if !rollappId.IsEIP155() {
+		return nil
+	}
+	// check to see if the RollappId has been registered before with same key
+	rollapp, isFound := k.GetRollappByEIP155(ctx, rollappId.GetEIP155ID())
+	// allow replacing EIP155 only when forking (previous rollapp is frozen)
+	if !isFound {
+		return nil
+	}
+	if !rollapp.Frozen {
+		return types.ErrRollappExists
+	}
+	previousRollappChainId, _ := types.NewChainID(rollapp.RollappId)
+	nextRevisionNumber := previousRollappChainId.GetRevisionNumber() + 1
+	if rollappId.GetRevisionNumber() != nextRevisionNumber {
```

Comment:
we probably should enforce the same chain name
to avoid `rollappX_100-1` forking into `rollappY_100-2`

Reply:
added name check